src/commands/scan.ts

```diff
@@ -0,0 +1,149 @@
+import {Command, flags} from '@oclif/command';
+import {Scanner, ScanResult} from '../scanner';
+import cli from 'cli-ux';
+import {RuleContext} from '../rules';
+import chalk = require('chalk');
+
+export default class Scan extends Command {
+  private _isVerbose = false;
+  private _isDebugMode = false;
+
+  static description =
+    'Scans azure resources for potential configuration issues';
+
+  static examples = [
+    `$ aza scan --scope <SCOPE>
+    [rule-name]
+        [✓ | ❌][rule-description]     
+        Resources:
+                [resource-ids]
+
+    [total-passing]
+    [total-failing]
+    [total-rules-scanned]   
+`,
+  ];
+
+  static flags = {
+    help: flags.help({char: 'h'}),
+    scope: flags.string({
+      char: 's',
+      description: 'azure subscription id to scan',
+    }),
+    group: flags.string({
+      description: 'azure subscription id to scan',
+      char: 'g',
+      multiple: true,
+      dependsOn: ['scope'],
+    }),
+    rule: flags.string({
+      char: 'r',
+      description: 'rules to execute',
+      multiple: true,
+    }),
+    dummy: flags.boolean({
+      char: 'd',
+      description: 'runs dummy rules to mock multi rule system',
+    }),
+    verbose: flags.boolean({
+      char: 'v',
+      description: 'prints all results',
+    }),
+    debug: flags.boolean({
+      description: 'prints debugging logs',
+    }),
+  };
+
+  private _logWithIndent(indentNum: number, msg: string) {
+    let indent = '';
+    if (indentNum) {
+      indent = '  ';
+      for (let i = indentNum; i > 0; i--) {
+        indent += indent;
+      }
+    }
+    this.log(indent + msg);
+  }
+
+  private _printResult(r: ScanResult) {
+    if (this._isVerbose || r.total) {
+      const description = chalk.grey(r.description);
```

Comment:
Will we get clean output from the CLI if we want to pipe it to the clipboard or a file? Or will it be filled with the terminal color/control characters?